doc/index.md

```diff
@@ -1032,68 +1050,153 @@ On success, returns the response [*headers*](#http.headers) and a [*stream*](#st
 
 ## http.server
 
-This interface is **unstable**.
+The http.server object is used to listen on sockets and respond to all types of HTTP requests. Each client request is threaded and independent of the other client requests. An `http.server` is instantiated by passing a number of options including an `onstream` response handler that processes the HTTP requests and provides a stream mechanism for response. `onstream` can also be used for testing and upgrading a request, with HTTP 1.1 to WebSockets being the notible example. The server can be made non-blocking when used in conjunction with cqueues. 
+
+### `server.listen(options)` <!-- --> {#http.server.listen}
+
+Create a new instance of an HTTP Server by passing in table `options`:
+
+  - `.client_timeout`: Timeout (in seconds) to wait for client to send first bytes and/or complete TLS handshake (default: 10)
+  - `.cq` (optional): A cqueues controller to use
+  - `.ctx`: An `openssl.ssl.context` object to use for tls connections. If `nil` is passed, a self-signed context will be generated
+  - `.family`: (optional) Protocol family
+  - `.host`: Address to bind to (required if not `.path`)
+  - `.port`: Port to bind to (optional if tls isn't `nil`, in which case defaults to 80 for `.tls == false` or 443 if `.tls == true`)
```

Comment:
Consider mentioning that `0` asks for any free port to be picked by the operating system